src/nvidia_resiliency_ext/inprocess/wrap.py

```diff
@@ -523,8 +561,8 @@ def __call__(self, fn, args, kwargs):
                             raise finalize_ex from term_ex
 
                         try:
-                            if wrapper.health_check is not None:
-                                wrapper.health_check(state.freeze())
+                            if wrapper.restart_health_check is not None:
+                                wrapper.restart_health_check(state.freeze())
```

Comment:
This will not raise any exception because the underlying routine doesn't do that.